ddtrace/opentelemetry/metric/meter_provider.go

```diff
@@ -0,0 +1,350 @@
+		return false
+	}
+
+	// If explicitly set, respect the value
+	metricsEnabled = strings.ToLower(strings.TrimSpace(metricsEnabled))
+	if metricsEnabled == "false" || metricsEnabled == "0" {
+		return false
+	}
+	if metricsEnabled == "true" || metricsEnabled == "1" {
+		return true
+	}
+
+	// Invalid value, default to disabled
+	return false
+}
+
+// Shutdown gracefully shuts down the MeterProvider, flushing any pending metrics.
+// For no-op providers, this is a no-op operation.
+func (mp *MeterProvider) Shutdown(ctx context.Context) error {
+	if mp.MeterProvider == nil || mp.isNoop {
+		return nil
+	}
+	return mp.MeterProvider.Shutdown(ctx)
+}
+
+// ForceFlush flushes any pending metrics.
+// For no-op providers, this is a no-op operation.
+func (mp *MeterProvider) ForceFlush(ctx context.Context) error {
+	if mp.MeterProvider == nil || mp.isNoop {
+		return nil
+	}
+	return mp.MeterProvider.ForceFlush(ctx)
+}
+
+// IsNoop returns true if this is a no-op MeterProvider that doesn't export metrics.
+func (mp *MeterProvider) IsNoop() bool {
+	return mp.isNoop
+}
+
+// deltaTemporalitySelector returns a temporality selector configured with Datadog defaults.
+// Default temporality is Delta, but non-monotonic instruments use Cumulative per OTel spec:
+// - Monotonic counters (Counter, ObservableCounter) → Delta (differences between measurements)
+// - Non-monotonic counters (UpDownCounter, ObservableUpDownCounter) → Cumulative (absolute values)
+// - Gauges (ObservableGauge) → Cumulative (point-in-time values)
+// - Histograms → Delta (distribution of measurements)
+// It respects OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE if set, with one exception:
+// UpDownCounter and ObservableUpDownCounter ALWAYS use Cumulative (even if DELTA is requested).
+func deltaTemporalitySelector() metric.TemporalitySelector {
+	// Check if user has explicitly set temporality preference
+	temporalityPref := strings.ToUpper(strings.TrimSpace(env.Get("OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE")))
+
+	return func(kind metric.InstrumentKind) metricdata.Temporality {
+		// UpDownCounter and Gauge ALWAYS use cumulative, regardless of preference
+		// They represent current state, not monotonic changes
+		switch kind {
+		case metric.InstrumentKindUpDownCounter,
+			metric.InstrumentKindObservableUpDownCounter,
+			metric.InstrumentKindObservableGauge:
+			return metricdata.CumulativeTemporality
+		}
+
+		// For monotonic instruments, respect the user's preference if set
+		if temporalityPref != "" {
+			switch temporalityPref {
+			case "CUMULATIVE":
+				return metricdata.CumulativeTemporality
+			case "DELTA":
+				return metricdata.DeltaTemporality
+			}
+		}
+
+		// Default behavior for monotonic instruments: Delta
+		switch kind {
+		case metric.InstrumentKindCounter,
+			metric.InstrumentKindHistogram,
+			metric.InstrumentKindObservableCounter:
+			return metricdata.DeltaTemporality
+		default:
+			return metricdata.DeltaTemporality
+		}
```

Comment:
This could be simplified to:
```suggestion
		return metricdata.DeltaTemporality
```

Is this what we want?